crates/compiler/builtins/roc/Json.roc

```diff
@@ -211,9 +211,18 @@ encodeString = \str ->
 
 encodeJsonEscapes : Str -> List U8
 encodeJsonEscapes = \str ->
-    Str.toUtf8 str
-    |> List.walk [] \bytes, byte ->
-        List.concat bytes (escapedByteToJson byte)
+    bytes = Str.toUtf8 str
+
+    # Reserve capacity for escaped bytes to reduce allocations
+    initial =
+        bytes
+        |> List.len
+        |> Num.mul 120
+        |> Num.divCeil 100
+        |> List.withCapacity
+
+    List.walk bytes initial \encodedBytes, byte ->
+        List.concat encodedBytes (escapedByteToJson byte)
```

Comment:
Potential idea for the future: since lots (most?) JSON payloads have no escapes, it might be nice to have a path where we don't allocate anything, and then as soon as we encounter the first thing that needs escaping, we allocate, copy in everything up to that point, and then continue from there.

Reply:
Done.